### PR TITLE
UCT/IB/MLX5: DEVX UAR return error

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9694/ for details (Mellanox internal link).

**Comment by Artemy-Mellanox** [comment]
bot:bgate:retest

**Comment by Artemy-Mellanox** [comment]
bot:bgate:retest

File: src/uct/ib/mlx5/ib_mlx5.c [hunk 1]
```diff
@@ -370,6 +370,9 @@ static ucs_status_t uct_ib_mlx5_devx_uar_init(uct_ib_mlx5_devx_uar_t *uar,
 {
 #if HAVE_DEVX
     uar->uar            = mlx5dv_devx_alloc_uar(md->super.dev.ibv_context, 0);
+    if (uar->uar == NULL) {
+        return UCS_ERR_NO_MEMORY;
```
Comment (brminich) [inline src/uct/ib/mlx5/ib_mlx5.c]: minor: maybe it is worth to print error/warn here, because this is the first error occurrence in UCX stack

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9727/ for details (Mellanox internal link).

File: src/uct/ib/mlx5/ib_mlx5.c (hunk 1 above)
Comment (shamisp) [reply src/uct/ib/mlx5/ib_mlx5.c]: I suggest to add some sort error or debug messages. It is royal pain to debug such low level failures.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9740/ for details (Mellanox internal link).

**Comment by Artemy-Mellanox** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9763/ for details (Mellanox internal link).

**Pull request merged**